ui/js/graphs.js

```diff
@@ -738,8 +754,53 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
         $scope.selectedProject = projects[0];
     }
     $scope.loadingTestData = false;
-
     var series = [];
+    // if we choose to add all related platforms, firstly we need
+    // to obtain all the platform of this project, then pick out the
+    // specific test by platform and project.
+    if(option === "addRelatedPlatforms") {
+        PhSeries.getAllSeries(relateSeries.relateProjectName,
+            $scope.timeRange, optionCollectionMap).then(
+            function(seriesData) {
+                var platformList = seriesData.platformList;
+                platformList.forEach(function(platform) {
+                    var testList = _.sortBy(_.filter(seriesData.seriesList,
+                    { platform: platform }), 'name');
+                    var temp = _.findWhere(testList, {"name": relateSeries.relateTest});
+                    // if found something different from the series we already have,
+                    // then we push it into the testsToAdd list.
+                    if (temp !== undefined && temp.signature !== relateSeries.signature) {
+                        $scope.testsToAdd.push(_.clone(temp));
+                    }
+                });
+                // tell people we find no related series on other platforms
+                if ($scope.testsToAdd.length === 0) {
+                    window.alert('Oops, there seems no other related platform about this series.');
+                    $modalInstance.close();
+                }
+            }
+        );
+    }
+    // if we choose to add all related branches, we should
+    // to find out all the branches of specific platform. Then query
+    // the test data one by one.
+    if(option === "addRelatedBranchs") {
+        $scope.projects.forEach(function(project) {
+            PhSeries.getAllSeries(project.name, $scope.timeRange, optionCollectionMap).then(
+                function(seriesData) {
+                    var testList = _.sortBy(_.filter(seriesData.seriesList,
+                        {platform: relateSeries.relatePlatform}), 'name');
+                    var temp = _.findWhere(testList, {"name": relateSeries.relateTest});
+                    if (temp !== undefined && temp.signature !== relateSeries.signature) {
+                        $scope.testsToAdd.push(_.clone(temp));
+                    }
+                });
+        });
+    //    if ($scope.testsToAdd.length === 0) {
+    //        window.alert('Oops, there seems no other related platform about this series.');
+    //        $modalInstance.close();
+    //    }
```

Comment:
Hm, I don't think so. Firstly, the variable also been called in here(https://github.com/mozilla/treeherder/pull/811/files?diff=unified#diff-5266f838afcedb1798e0cf4a88cba462R777) and $scope.testsToAdd variable been defined in line 770 (https://github.com/mozilla/treeherder/blob/master/ui/js/graphs.js#L770). So it should also tells me  "$scope.testsToAdd" is not defined when I call add related platforms, but it haven't.

Secondly, I had try to move $scope.testsToAdd before I call it. Of course, it pass and show me no error. But I got some different behaviors from 'add related platforms'.
When I call 'add related platforms', the tests chooser view will comes out first and show me the alert window after. But when I call 'add related branched', the alert window will comes out first and tests chooser view will shows up after I click the 'ok' button.
The alert window shows up first when I want to add all related branches
<img width="1278" alt="screen shot 2015-07-28 at 11 32 19 am" src="https://cloud.githubusercontent.com/assets/2166227/8923147/8ac1fd0c-351f-11e5-89f6-7d3df6683fab.png">
But add related platforms it will shows up after test chooser
<img width="1277" alt="screen shot 2015-07-28 at 11 32 37 am" src="https://cloud.githubusercontent.com/assets/2166227/8923166/b806eebc-351f-11e5-9c07-f97700c39c35.png">